autosklearn/estimators.py

```diff
@@ -655,19 +665,27 @@ def fit_ensemble(
             is independent of the ``ensemble_class`` argument and this
             pruning step is done prior to constructing an ensemble.
 
-        ensemble_class : Type[AbstractEnsemble], optional (default=EnsembleSelection)
+        ensemble_class : Type[AbstractEnsemble] | "default", optional (default="default")
             Class implementing the post-hoc ensemble algorithm. Set to
-            ``None`` to disable ensemble building or use ``SingleBest``
+            ``None`` to disable ensemble building or use class:`SingleBest`
             to obtain only use the single best model instead of an
             ensemble.
 
-        metrics: Scorer | Sequence[Scorer] | None = None
+            If set to "default" it will use :class:`EnsembleSelection` for
+            single-objective problems and :class:`MultiObjectiveDummyEnsemble`
+            for multi-objective problems.
+
+        metric: Scorer | Sequence[Scorer] | None = None
             A metric or list of metrics to score the ensemble with
 
         Returns
         -------
         self
-        """
+        """  # noqa: E501
+
+        # Need to resolve the ensemble class here so we can act on it below.
+        ensemble_class = self.resolve_ensemble_class(ensemble_class, metric)
```

Comment:
Fixed an issue here with regards to what happens when `metric = None`, see the comment in the commits